test/e2e/framework/util.go

```diff
@@ -2066,15 +2066,28 @@ func AssertCleanup(ns string, selectors ...string) {
 	if ns != "" {
 		nsArg = fmt.Sprintf("--namespace=%s", ns)
 	}
-	for _, selector := range selectors {
-		resources := RunKubectlOrDie("get", "rc,svc", "-l", selector, "--no-headers", nsArg)
-		if resources != "" {
-			Failf("Resources left running after stop:\n%s", resources)
-		}
-		pods := RunKubectlOrDie("get", "pods", "-l", selector, nsArg, "-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}{{ \"\\n\" }}{{ end }}{{ end }}")
-		if pods != "" {
-			Failf("Pods left unterminated after stop:\n%s", pods)
+
+	backoff := wait.Backoff{
+		Duration: 5 * time.Second,
+		Factor:   2,
+		Steps:    3,
+	}
+	verifyCleanupFunc := func() (bool, error) {
+		for _, selector := range selectors {
+			resources := RunKubectlOrDie("get", "rc,svc", "-l", selector, "--no-headers", nsArg)
+			if resources != "" {
+				return false, fmt.Errorf("Resources left running after stop:\n%s", resources)
```

Comment:
You should not return an error here, or the backoff func will stop earlier. See: https://github.com/kubernetes/apimachinery/blob/210bcde8e5a7715fe776460ca1df1a3f2738725e/pkg/util/wait/wait.go#L203
But you may want to keep the error somewhere.
I'd expect something like:
```
    var e error
	verifyCleanupFunc := func() (bool, error) {
        e = nil
		for _, selector := range selectors {
			resources := RunKubectlOrDie("get", "rc,svc", "-l", selector, "--no-headers", nsArg)
			if resources != "" {
                 e = fmt.Errorf("Resources left running after stop:\n%s", resources)
				return false, nil
			}
			pods := RunKubectlOrDie("get", "pods", "-l", selector, nsArg, "-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}{{ \"\\n\" }}{{ end }}{{ end }}")
			if pods != "" {
                 e = fmt.Errorf("Pods left unterminated after stop:\n%s", pods)
				return false, nil
			}
		}
		return true, nil
	}
	err := wait.ExponentialBackoff(backoff, verifyCleanupFunc)
	if err != nil {
         // if timeout, you want to return e or err + e.
         // err itself may not be very helpful.
		Failf(e.Error())
	}
```

Reply:
That's a good catch and thank you. I updated it.